packages/stark-core/src/session/services/session.service.ts

```diff
@@ -0,0 +1,411 @@
+					map((session: StarkSession) => {
+						if (typeof session.user === "undefined") {
+							// reject transition in case there is no user in the session
+							throw new Error(starkUnauthenticatedUserError);
+						} else {
+							return true;
+						}
+					})
+				).toPromise<boolean>();
+			},
+			{priority: 1000} // very high priority (this hook should be the first one to be called to reject transitions immediately)
+		);
+	}
+
+	/**
+	 * Performs all the necessary actions to initialize the session.
+	 * It dispatches a INITIALIZE_SESSION action to the NGRX-Store
+	 * @param user - The user used to initialize the session.
+	 */
+	protected initializeSession(user: StarkUser): void {
+		this.store.dispatch(new InitializeSession(user));
+		this.startIdleService();
+		this.startKeepaliveService();
+		this.store.dispatch(new InitializeSessionSuccess());
+	}
+
+	/**
+	 * Performs all the necessary actions to destroy the session. The user stored in the session is removed.
+	 * It dispatches a DESTROY_SESSION action to the NGRX-Store
+	 */
+	protected destroySession(): void {
+		this.store.dispatch(new DestroySession());
+		this.stopIdleService();
+		this.stopKeepaliveService();
+		this.store.dispatch(new DestroySessionSuccess());
+	}
+
+	public login(user: StarkUser): void {
+		this.initializeSession(user);
+	}
+
+	public logout(): void {
+		// dispatch action so an effect can run any logic if needed
+		this.store.dispatch(new SessionLogout());
+		// the session will always be destroyed right after the response of the logout HTTP call (regardless of its result)
+		this.sendLogoutRequest(this.appConfig.logoutUrl, "", true).subscribe(
+			() => this.destroySession(),
+			() => this.destroySession()
+		);
+	}
+
+	protected sendLogoutRequest(url: string, serializedData: string, async: boolean = true): Observable<void> {
+		const httpRequest$: Subject<void> = new Subject<void>();
+
+		const emitXhrResult: Function = (xhrRequest: XMLHttpRequest) => {
+			if (xhrRequest.readyState === XMLHttpRequest.DONE) {
+				if (xhrRequest.status === StarkHttpStatusCodes.HTTP_200_OK || xhrRequest.status === StarkHttpStatusCodes.HTTP_201_CREATED) {
+					httpRequest$.next();
+					httpRequest$.complete();
+				} else {
+					httpRequest$.error(xhrRequest.status);
+				}
+			}
+		};
+
+		const xhr: XMLHttpRequest = new XMLHttpRequest();
+
+		if (async) {
+			xhr.onreadystatechange = () => {
+				emitXhrResult(xhr);
+			};
+		} else {
+			emitXhrResult(xhr);
+		}
+
+		// catch any error raised by the browser while opening the connection. for example:
+		// Chrome "mixed content" error: https://developers.google.com/web/fundamentals/security/prevent-mixed-content/what-is-mixed-content
+		// IE "Access is denied" error: https://stackoverflow.com/questions/22098259/access-denied-in-ie-10-and-11-when-ajax-target-is-localhost
+		try {
+			xhr.open("GET", url, async);
+			// FIXME Uncomment when XSRF Service is implemented
+			// this.xsrfService.configureXHR(xhr);
```

Comment:
You can remove this lne of code, at the end the XSRF service should not be used here because it is a GET request (XSRF only makes sense for state-changing requests (PUT, POST, PATCH and DELETE). So you can remove completely the references to the XSRF service from here.

I already did this change in the old Stark... ;)